client/admin/components/users/ImportDialog.vue

```diff
@@ -0,0 +1,168 @@
+<template>
+  <v-dialog v-hotkey="{ esc: close }" v-model="visible" width="700">
+    <v-btn slot="activator" color="blue-grey" outline>
+      <v-icon>cloud_upload</v-icon>Import
+    </v-btn>
+    <v-form @submit.prevent="save">
+      <v-card class="pa-3">
+        <v-card-title class="headline">
+          Import Users
+        </v-card-title>
+        <v-card-text>
+          <v-text-field
+            ref="textFile"
+            v-model="filename"
+            :error-messages="showErrors"
+            :disabled="importing"
+            @click="$refs.fileInput.click()"
+            prepend-icon="attach_file"
+            label="Upload .xls, .xlsx or .csv file"
+            readonly
+            single-line/>
+          <input
+            v-validate="rules"
+            v-show="false"
+            ref="fileInput"
+            @change="onFileSelected"
+            data-vv-name="file"
+            type="file">
+        </v-card-text>
+        <v-card-text v-show="importing" class="loader-container">
+          <circular-progress :width="40" :height="40"/>
+        </v-card-text>
+        <v-card-actions v-show="!importing">
+          <v-spacer/>
+          <v-fade-transition>
+            <v-btn
+              v-show="visible && errors"
+              @click="downloadErrorsFile"
+              color="error">
+              <v-icon>cloud_download</v-icon>Errors
+            </v-btn>
+          </v-fade-transition>
+          <v-btn @click="close">Cancel</v-btn>
+          <v-btn :disabled="disabled" color="success" type="submit">
+            Import
+          </v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-form>
+  </v-dialog>
+</template>
+
+<script>
+import api from '@/admin/api/user';
+import CircularProgress from '@/common/components/CircularProgress';
+import JSZip from 'jszip';
+import saveAs from 'save-as';
+import { withFocusTrap } from '@/common/focustrap';
+import { withValidation } from '@/common/validation';
+
+const el = vm => vm.$children[0].$refs.dialog;
+const rules = {
+  required: true,
+  mimes:
+  [
+    'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+    'application/vnd.ms-excel',
+    'text/csv'
+  ]
```

Comment:
[Check point No 3](https://blog.codinghorror.com/new-programming-jargon/) and apply it here on array brackets :wink: